src/commands/send-many-memo-safe.ts

```diff
@@ -0,0 +1,284 @@
+    }),
+    broadcast: flags.boolean({
+      char: 'b',
+      default: false,
+      description:
+        'Whether to broadcast this transaction. Omitting this flag will not broadcast the transaction.',
+    }),
+    network: flags.string({
+      char: 'n',
+      description: 'Which network to broadcast this to',
+      options: ['mocknet', 'testnet', 'mainnet'],
+      default: 'testnet',
+    }),
+    nodeUrl: flags.string({
+      required: false,
+      char: 'u',
+      description:
+        'A default node URL will be used based on the `network` option. Use this flag to manually override.',
+    }),
+    quiet: flags.boolean({
+      char: 'q',
+      default: false,
+      description: `
+Reduce logging from this command. If this flag is passed with the broadcast (-b) flag,
+only the transaction ID will be logged. If the quiet flagged is passed without broadcast, 
+only the raw transaction hex will be logged.
+`,
+    }),
+    jsonOutput: flags.boolean({
+      char: 'j',
+      default: false,
+      description: 'Output data in JSON format',
+    }),
+    contractAddress: flags.string({
+      char: 'c',
+      description:
+        'Manually specify the contract address for send-many-memo. If omitted, default contracts will be used.',
+    }),
+    nonce: flags.integer({
+      description: 'Optionally specify a nonce for this transaction',
+    }),
+  };
+
+  static args = [
+    {
+      name: 'recipients',
+      description: `
+A set of recipients in the format of "address,amount_ustx,memo". Memo is optional.
+Example: STADMRP577SC3MCNP7T3PRSTZBJ75FJ59JGABZTW,100,memo ST2WPFYAW85A0YK9ACJR8JGWPM19VWYF90J8P5ZTH,50
+      `,
+    },
+  ];
+
+  getNetwork() {
+    const { flags } = this.parse(SendManyMemoSafe);
+    const networks = {
+      mainnet: StacksMainnet,
+      testnet: StacksTestnet,
+      mocknet: StacksMocknet,
+    };
+
+    return networks[flags.network as NetworkString];
+  }
+
+  getContract(network: StacksNetwork) {
+    return network.chainId === ChainID.Mainnet
+      ? DEFAULT_MAINNET_CONTRACT
+      : DEFAULT_TESTNET_CONTRACT;
+  }
+
+  async run() {
+    const { argv, flags } = this.parse(SendManyMemoSafe);
+
+    const recipients: Recipient[] = argv.map(arg => {
+      const [address, amount, memo] = arg.split(',');
+      if (!validateStacksAddress(address)) {
+        throw new Error(`${address} is not a valid STX address`);
+      }
+      if (!isNormalInteger(amount)) {
+        throw new Error(`${amount} is not a valid integer.`);
+      }
+      return {
+        address,
+        amount,
+        memo,
+      };
+    });
+
+    const networkClass = this.getNetwork();
+    if (!networkClass) {
+      throw new Error('Unable to get network');
+    }
+    const network = new networkClass();
+    if (flags.nodeUrl) {
+      network.coreApiUrl = flags.nodeUrl;
+    }
+
+    const memoExpectedRecipients = await checkMemoExpected(network, recipients);
+    if (memoExpectedRecipients.length) {
```

Comment:
Maybe I've been staring at Rust code for too long, but shouldn't this be `memoExpectedRecipients.length > 0`?

Reply:
A non-zero integer is truthy in JavaScript 😁. We can make it explicit though.